src/modules/fancyzones/lib/JsonHelpers.h

```diff
@@ -0,0 +1,244 @@
+#pragma once
+
+#include <common/settings_helpers.h>
+#include <common/json.h>
+
+#include <string>
+#include <strsafe.h>
+#include <unordered_map>
+#include <variant>
+#include <optional>
+#include <vector>
+#include <winnt.h>
+
+namespace JSONHelpers
+{
+    constexpr int MAX_ZONE_COUNT = 50;
+
+    enum class ZoneSetLayoutType : int
+    {
+        Focus = 0,
+        Columns,
+        Rows,
+        Grid,
+        PriorityGrid,
+        Custom
+    };
+
+    enum class CustomLayoutType : int
+    {
+        Grid = 0,
+        Canvas
+    };
+
+    std::wstring TypeToString(ZoneSetLayoutType type);
+    ZoneSetLayoutType TypeFromString(const std::wstring& typeStr);
+
+    ZoneSetLayoutType TypeFromLayoutId(int layoutID);
+    int TypeToLayoutId(JSONHelpers::ZoneSetLayoutType layoutType);
+
+    using TZoneCount = int;
+    using TZoneUUID = std::wstring;
+    using TAppPath = std::wstring;
+    using TDeviceID = std::wstring;
+
+    struct CanvasLayoutInfo
+    {
+        int referenceWidth;
+        int referenceHeight;
+        struct Rect
+        {
+            int x;
+            int y;
+            int width;
+            int height;
+        };
+        std::vector<CanvasLayoutInfo::Rect> zones;
+
+        static json::JsonObject ToJson(const CanvasLayoutInfo& canvasInfo);
+        static std::optional<CanvasLayoutInfo> FromJson(const json::JsonObject& infoJson);
+    };
+
+    class GridLayoutInfo
+    {
+    public:
+        struct Minimal
+        {
+            int rows;
+            int columns;
+        };
+
+        struct Full
+        {
+            int rows;
+            int columns;
+            const std::vector<int>& rowsPercents;
+            const std::vector<int>& columnsPercents;
+            const std::vector<std::vector<int>>& cellChildMap;
+        };
+
+        GridLayoutInfo(const Minimal& info);
```

Comment:
Is there a reason to have these two structures (`Minimal` and `Full`), they are exposed outside of `GridLayoutInfo` class, but it's not really clear what is their purpose. It looks more clean to me to simply have two constructors, having different number of arguments (and maybe default constructor).

Reply:
The reason was to keep args in constructors readable here.
https://github.com/stefansjfw/PowerToys/blob/f751b8542346f963d042e8c8b545948c5caf1ded/src/modules/fancyzones/lib/ZoneSet.cpp#L20

Without structures you see only bunch of numbers and it's hard to realize what is what there. And also there is possibility to mix up arguments.